sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ThresholdBasedAvailabilityStrategy.java

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * The type Threshold based retry availability strategy.
+ */
+public final class ThresholdBasedAvailabilityStrategy extends AvailabilityStrategy {
+    private static final Duration DEFAULT_THRESHOLD = Duration.ofMillis(500);
+    private static final Duration DEFAULT_THRESHOLD_STEP = Duration.ofMillis(100);
+    private Duration threshold;
```

Comment:
Do we want `threshold` and `thresholdStep` to be mutable?

Reply:
Made them immutable for now.